src/EditorFeatures/Core/Implementation/RenameTracking/RenameTrackingTaggerProvider.TrackingSession.cs

```diff
@@ -73,11 +73,15 @@ public TrackingSession(StateMachine stateMachine, SnapshotSpan snapshotSpan, IAs
 
                     var asyncToken = _asyncListener.BeginAsyncOperation(GetType().Name + ".UpdateTrackingSessionAfterIsRenamableIdentifierTask");
 
-                    _isRenamableIdentifierTask.SafeContinueWith(
-                        t => stateMachine.UpdateTrackingSessionIfRenamable(),
+                    _isRenamableIdentifierTask.SafeContinueWithFromAsync(
+                        async t =>
+                        {
+                            await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(_cancellationToken);
```

Comment:
❕ This needs to yield if called from the main thread in order to preserve the original behavior.